sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -415,39 +415,66 @@ opentitan_functest(
+        srcs = ["chip_specific_startup.c"],
+        args = [],
+        cw310 = cw310_params(
+            bitstream = ":bitstream_chip_specific_startup_{}".format(lc_state.lower()),
+            otp = ":otp_img_e2e_bootstrap_entry_{}".format(lc_state.lower()),
+            tags = ["vivado"] + maybe_skip_in_ci(lc_state_val),
+            test_cmds = [
+                "--rom-kind=rom",
+                "--bitstream=\"$(location //sw/device/silicon_creator/rom/e2e:bitstream_chip_specific_startup_{})\"".format(lc_state.lower()),
+                "--bootstrap=\"$(location {flash})\"",
+                "--otp-unprogrammed",
+            ],
+        ),
+        signed = True,
+        targets = ["cw310_rom"],
+        test_harness = "//sw/host/tests/rom/e2e_chip_specific_startup",
+        deps = [
+            "//hw/ip/csrng/data:csrng_regs",
+            "//hw/ip/edn/data:edn_regs",
+            "//hw/ip/entropy_src/data:entropy_src_regs",
+            "//hw/ip/otp_ctrl/data:otp_ctrl_regs",
+            "//hw/top_earlgrey/ip/sensor_ctrl/data:sensor_ctrl_regs",
+            "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+            "//sw/device/lib/base:mmio",
+            "//sw/device/lib/dif:clkmgr",
+            "//sw/device/lib/dif:lc_ctrl",
+            "//sw/device/lib/dif:otp_ctrl",
+            "//sw/device/lib/dif:sram_ctrl",
+            "//sw/device/lib/runtime:log",
+            "//sw/device/lib/testing/json:chip_specific_startup",
+            "//sw/device/lib/testing/json:command",
+            "//sw/device/lib/testing/test_framework:ottf_main",
+            "//sw/device/lib/testing/test_framework:ujson_ottf",
+            "//sw/device/lib/ujson",
         ],
-    ),
-    signed = True,
-    targets = ["cw310_rom"],
-    test_harness = "//sw/host/tests/rom/e2e_chip_specific_startup",
-    deps = [
-        "//hw/ip/csrng/data:csrng_regs",
-        "//hw/ip/edn/data:edn_regs",
-        "//hw/ip/entropy_src/data:entropy_src_regs",
-        "//hw/ip/otp_ctrl/data:otp_ctrl_regs",
-        "//hw/top_earlgrey/ip/sensor_ctrl/data:sensor_ctrl_regs",
-        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
-        "//sw/device/lib/base:mmio",
-        "//sw/device/lib/dif:clkmgr",
-        "//sw/device/lib/dif:lc_ctrl",
-        "//sw/device/lib/dif:otp_ctrl",
-        "//sw/device/lib/dif:sram_ctrl",
-        "//sw/device/lib/runtime:log",
-        "//sw/device/lib/testing/json:chip_specific_startup",
-        "//sw/device/lib/testing/json:command",
-        "//sw/device/lib/testing/test_framework:ottf_main",
-        "//sw/device/lib/testing/test_framework:ujson_ottf",
-        "//sw/device/lib/ujson",
+    )
+    for lc_state, lc_state_val in structs.to_dict(CONST.LCV).items()
+]
+
+test_suite(
+    name = "e2e_chip_specific_startup",
+    tags = ["manual"],
+    tests = [
+        "e2e_chip_specific_startup_{}".format(lc_state.lower())
+        for lc_state in structs.to_dict(CONST.LCV)
```

Comment:
@dmcardle can you remind me if the macro for this use-case is already merged?

Reply:
Oh I see, I didn't know we had that. Fixed.

Tbh though, using this macro only really saves ~20 chars per instance at the expense of some complexity understanding what it does, where before things were more verbose. However, I suppose the real value come from the ability of this macro to do some filtering.

Reply:
Yep, the idea was that it enables filtering without relying on magic string constants at the call-site.